unlock-app/src/services/WaasProvider.ts

```diff
@@ -95,6 +95,7 @@ export default class WaasProvider extends ethers.JsonRpcProvider {
   }
 
   async disconnect() {
+    localStorage.removeItem('nextAuthProvider')
```

Comment:
You are doing this in another place. That is confusing. Can we do it in a single place?